pipelines/curate_preferences.py

```
@@ -0,0 +1,603 @@
+            with os.fdopen(descriptor, "w", encoding="utf-8", newline="\n") as handle:
+                handle.write(payload)
+                handle.flush()
+                os.fsync(handle.fileno())
+    except Exception:
+        # Remove only files this invocation created; pre-existing paths are
+        # rejected before and during O_EXCL creation.
+        for path in created:
+            try:
+                path.unlink()
+            except FileNotFoundError:
+                pass
```

Ruff (0.16.1)

[warning] Use `contextlib.suppress(FileNotFoundError)` instead of `try`-`except`-`pass`

Replace `try`-`except`-`pass` with `with contextlib.suppress(FileNotFoundError): ...`

(SIM105)